operator/v1/types_ingress.go

```diff
@@ -253,6 +253,20 @@ type IngressControllerSpec struct {
 	HTTPCompression HTTPCompressionPolicy `json:"httpCompression,omitempty"`
 }
 
+// +kubebuilder:validation:Enum:=Normal;Debug;Trace
+type IngressLogLevel string
+
+var (
+	// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+	IngressLogLevelNormal IngressLogLevel = "Normal"
+
+	// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+	IngressLogLevelDebug IngressLogLevel = "Debug"
+
+	// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+	IngressLogLevelTrace IngressLogLevel = "Trace"
+)
```

Comment:
The EP also has:
```
	// TraceAll is used when something is broken at the level of API content/decoding.  It will dump complete body content.  If you turn this on in a production cluster
	// prepare from serious performance issues and massive amounts of logs.  In kube, this is probably glog=8.
	TraceAll LogLevel = "TraceAll"
```

Reply:
good catch; I'll add that.

Reply:
This is in the same package we use for our other operators.  Matching the OperatorSpec seems valuable even without embedding the entire type. See https://github.com/openshift/api/blob/master/operator/v1/types.go#L55-L71